File: data-structures/3_queue/queue.ts
class Queue<T> {
  items: Array<T> = [];

  enqueue(value: T) {
    this.items.push(value);
  }

  dequeue() {
    return this.items.shift();
  }

  peek() {
    return this.items[0];
  }

  isEmpty() {
    return this.items.length === 0;
  }

  size() {
    return this.items.length;
  }
}

const queue = new Queue<number>();
queue.enqueue(12);
queue.enqueue(24);
queue.enqueue(36);

queue.dequeue();

console.log(queue.peek());

File: data-structures/3_queue/queue.py
class Queue:
    def __init__(self):
        self.items = []
    
    def enqueue(self, value):
        self.items.append(value)
    
    def dequeue(self):
        return self.items.pop(0)
    
    def peek(self):
        return self.items[0]
    
    def isEmpty(self):
        return len(self.items) == 0
    
    def size(self):
        return len(self.items)
    
queue = Queue()
queue.enqueue(5)
queue.enqueue(6)
queue.enqueue(7)
queue.enqueue(8)
queue.enqueue(9)

print(queue.dequeue())
print(queue.peek())